src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java

```diff
@@ -0,0 +1,177 @@
+
+                    for (int i = 0; i < simSources.size(); i++)
+                    {
+                        SourcePacketExtension spe = simSources.get(i);
+                        long primarySSRC = spe.getSSRC();
+                        grouped.add(primarySSRC);
+
+                        long rtxSSRC = -1;
+                        if (rtxPairs.containsKey(primarySSRC))
+                        {
+                            rtxSSRC = rtxPairs.remove(primarySSRC);
+                            grouped.add(rtxSSRC);
+                        }
+
+                        int subjectiveQualityIdx = i * 3;
+                        RTPEncodingImpl encoding = new RTPEncodingImpl(
+                            track, subjectiveQualityIdx, primarySSRC, rtxSSRC,
+                            0, null);
+
+                        rtpEncodings[subjectiveQualityIdx] = encoding;
+
+                        subjectiveQualityIdx = i * 3 + 1;
+                        encoding = new RTPEncodingImpl(
+                            track, subjectiveQualityIdx, primarySSRC, rtxSSRC,
+                            1, new RTPEncodingImpl[] { encoding });
+                        rtpEncodings[subjectiveQualityIdx] = encoding;
+
+                        subjectiveQualityIdx = i * 3 + 2;
+                        encoding = new RTPEncodingImpl(
+                            track, subjectiveQualityIdx, primarySSRC, rtxSSRC,
+                            2, new RTPEncodingImpl[] { encoding });
+                        rtpEncodings[subjectiveQualityIdx] = encoding;
+                    }
+
+                    tracks.add(track);
+                }
+            }
+
+            if (!rtxPairs.isEmpty())
+            {
+                for (Map.Entry<Long, Long> fidEntry : rtxPairs.entrySet())
+                {
+                    Long primarySSRC = fidEntry.getKey();
+                    Long rtxSSRC = fidEntry.getValue();
+
+                    RTPEncodingImpl[] encodings = new RTPEncodingImpl[1];
+                    MediaStreamTrackImpl track = new MediaStreamTrackImpl(
+                        mediaStreamTrackReceiver, encodings);
+                    encodings[0]
+                        = new RTPEncodingImpl(track, primarySSRC, rtxSSRC);
+
+                    grouped.add(primarySSRC);
+                    grouped.add(rtxSSRC);
+                    tracks.add(track);
+                }
+            }
+        }
+
+        if (hasSources)
+        {
+            for (SourcePacketExtension spe : sources)
+            {
+                long mediaSSRC = spe.getSSRC();
+
+                if (!grouped.contains(mediaSSRC))
+                {
+                    RTPEncodingImpl[] encodings = new RTPEncodingImpl[1];
+                    MediaStreamTrackImpl track = new MediaStreamTrackImpl(
+                        mediaStreamTrackReceiver, encodings);
+                    encodings[0] = new RTPEncodingImpl(track, mediaSSRC);
```

Comment:
Same comment as above -- if we don't care about temporal layers, the RTPEncoding should not care about temporal layers.